src/orchard/misc.clj

```diff
@@ -20,16 +20,27 @@
   []
   (not (nil? (boot-fake-classpath))))
 
+(defn url?
+  "Check whether the argument is an url"
+  [u]
+  (instance? java.net.URL u))
+
 (defn directory?
-  "Whether the argument is a directory"
+  "Whether the argument is a directory or an url that points to a directory"
   [f]
-  (.isDirectory (io/as-file f)))
+  (if (url? f)
+    (and (= (.getProtocol ^java.net.URL f) "file")
```

Comment:
Here you can pull this `and` higher.

Reply:
You mean like this ?
```
(and (url? f)
        (= (.getProtocol ^java.net.URL f) "file")
        (.isDirectory (io/as-file f)))
```
Then it would not check nonURLs correctly?

Reply:
I was thinking of something like:

`(and (url? f) (= (.getProtocol ^java.net.URL f) "file")) ...`

With a more complex conditional you can even get rid of the duplicated .isDirectory check, but this will probably make the code hard to read.

Reply:
Sorry, but i still do not understand.
You mean like this?
```
(defn directory?
  "Whether the argument is a directory or an url that points to a directory"
  [f]
  (if (and (url? f) (= (.getProtocol ^java.net.URL f) "file"))
    (.isDirectory (io/as-file f))))
```

Reply:
My bad, I just realized where your confusion was coming from. If we structure the condition in this manner, the else branch won't be correct. I should stop reviewing code while doing something else. :D

Reply:
No problem